dcc-portal-ui/app/scripts/genesets/js/genesets.js

```diff
@@ -247,15 +250,78 @@
                 n.geneId = ensemblId;
               });
             });
+            
+            SetService.getTransientSet('GENE', {
+              'filters': {gene:{pathwayId:{is:[parentPathwayId]}}},
+              'sortBy': 'id',
+              'sortOrder': 'ASCENDING',
+              'name': parentPathwayId,
+              'size': _ctrl.geneSet.geneCount,
+              'transient': true 
+            }).then(function (entitySetData) {
+              CompoundsService.getCompoundsFromEntitySetId(entitySetData.id).then(function(drugs) {                
+                var drugMap = {};
+                _.forEach(drugs, function(drug) {
+                  var zincId = drug.zincId;
+                  var name = drug.name;
+                  var genes = drug.genes;
+                  _.forEach(genes, function(gene) {
+                    var uniprotId = gene.uniprot;
+                    
+                    if (_.isUndefined(drugMap[uniprotId])) {
+                      drugMap[uniprotId] = [];
+                    }
+                    
+                    drugMap[uniprotId].push({
+                      'zincId': zincId,
+                      'name': name
+                    });
+                  });
+                });
 
-            _ctrl.pathway.highlights = pathwayHighlights;
+                var pathwayDrugHighlights = [];
+                // Normalize into array
+                _.forEach(map,function(value,id) {
+                  if(value && value.dbIds && drugMap[id]) {
```

Comment:
This check shouldn't have to exist, the reason that it does is because you're getting bad data right?
Not your fault, but `map` shouldn't be the value that it is, map is the Restangular response, and it really should be `map.plain()` to get rid of the extra restangular keys on the object